engine/runtime/src/main/java/org/enso/interpreter/runtime/type/Types.java

```diff
@@ -130,6 +131,8 @@ public static String getName(Object value) {
       return ConstantsGen.MANAGED_RESOURCE;
     } else if (TypesGen.isArray(value)) {
       return ConstantsGen.ARRAY;
+    } else if (TypesGen.isArrayOverBuffer(value)) {
+      return ConstantsGen.ARRAY_OVER_BUFFER;
```

Comment:
Ideally we'd have these merged but I'm not sure if it is safe to do?

OTOH, as long as we use the ArrayOverBuffer only in internal functions, it probably doesn't matter, because we don't check the type anyway, _what can go wrong?_ (_famous last words_)
Still this type may get used in more places in the future so I'd prefer to do it the _right way_, but I'm not sure which way it is? cc: @JaroslavTulach @hubertp

Reply:
Discussed with @JaroslavTulach and collapsed the two array representations to report the same Enso type.